app/controllers/mass_rollouts_controller.rb

```diff
@@ -16,23 +16,37 @@ def create
   end
 
   def deploy
-    environment = deploy_group.environment
-    template_stages = environment.template_stages.all
-    missing_only = params[:missing_only] == "true"
-    stages_to_deploy = missing_only ? deploy_group.stages.reject(&:last_successful_deploy) : deploy_group.stages
-    deploys = stages_to_deploy.map do |stage|
-      template_stage = template_stages.detect { |ts| ts.project_id == stage.project.id }
-      next unless template_stage
+    successful, missing = deploy_group.stages.partition(&:last_successful_deploy)
+    stages_to_deploy = []
+
+    if params[:successful] == "true"
+      stages_to_deploy += successful.map { |stage| [stage, stage.last_successful_deploy.reference] }
```

Comment:
this is changing the behavior ... before it was deploying the template stages last deploy

Reply:
which is the indented way this should work since it's assuming the template stage is up to date

Reply:
/cc @zendesk/bre for input on if the behaviour change is okay or if there's a use-case/need to keep overwriting the last deploy with what is on the template stage. to me, it seems more risky if the template stage is out-of-sync somehow.